docs/getting-started.md

```diff
@@ -0,0 +1,99 @@
+
+#### Log in to OpenShift
+
+First off, make sure you are logged in to an OpenShift cluster. The easiest way to get a single node OpenShift cluster in by using [minishift](https://docs.openshift.org/latest/minishift/index.html), but `ocdev` will work with any other OpenShift instance you are logged in to.
+
+Make sure you are logged in to the cluster by running `oc whoami` command.
+
+#### Create an application
+
+An application is the umbrella under which you will build all the components (microservices) of your entire application.
+
+Let's create an application -
+
+```console
+$ ocdev application create nodeapp 
+Creating application: nodeapp
+Switched to application: nodeapp
+```
+
+#### Create a component
+
+Now that you have created an application, now add a component of type _nodejs_ to the application, from the current directory where our code lies.
+
+```console
+$ ocdev component create nodejs --dir=. 
+--> Found image 2809a54 (3 weeks old) in image stream "openshift/nodejs" under tag "6" for "nodejs"
+--> Creating resources with label app=nodeapp,app.kubernetes.io/component-name=nodejs,app.kubernetes.io/name=nodeapp ...
+    imagestream "nodejs" created       
+    buildconfig "nodejs" created       
+    deploymentconfig "nodejs" created  
+    service "nodejs" created           
+--> Success        
+    Build scheduled, use 'oc logs -f bc/nodejs' to track its progress.
+    Application is not exposed. You can expose services to the outside world by executing one or more of the commands below:
+     'oc expose svc/nodejs'            
+    Run 'oc status' to view your app.  
+
+please wait, building application...   
+Uploading directory "." as binary input for the build ...
+build "nodejs-2" started               
+Pushing image 172.30.1.1:5000/myproject/nodejs:latest ...
+Push successful
+```
+
+Great news! Your component has been deployed on OpenShift now. Let's quickly check how it looks!
+
+##### Connect to the component
+
+We need to expose the component as a route so we can connect to it.
+
+Get the name of the service by running -
+```console
+$ oc get services
+NAME      CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+nodejs    172.30.241.15   <none>        8080/TCP   9m
+```
+
+Now we need to expose the service to create a route -
+```console
+$ oc expose service nodejs
+route "nodejs" exposed
+
+$ oc get route
+NAME      HOST/PORT                                PATH      SERVICES   PORT       TERMINATION   WILDCARD
+nodejs    nodejs-myproject.192.168.42.147.nip.io             nodejs     8080-tcp                 None
+```
+
+Now just open the URL `nodejs-myproject.192.168.42.147.nip.io` in the browser and you will be able to view your deployed application.
+
+#### Push new changes to the component
+
+Well, your application looks great, but now you've made some changes in the code. Let's deploy these changes and see how it looks.
+
+```console
+$ ocdev component push nodejs
```

Comment:
this might be good place to explain what current component is and use just `ocdev component push`

Reply:
done